Dockerfile

```diff
@@ -1,35 +1,70 @@
-FROM python:3.9.7-slim@sha256:cd1045dbabff11dab74379e25f7974aa7638bc5ad46755d67d0f1f1783aee101
+# syntax=docker/dockerfile:1.3
 
-ENV PYTHONUNBUFFERED=1 \
-    PYTHONPATH=/app/ \
-    PORT=8888
+FROM python:3.9-slim-bullseye as compile
 
-EXPOSE $PORT
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends \
+    # Needed to download Rust
+    curl \
+    # Needed to build psycopg and uWSGI 
+    build-essential \
+    python-dev \
+    libpq-dev \
+    # Needed to build uwsgi-dogstatsd plugin
+    git
 
-# add a non-privileged user for installing and running
-# the application
-RUN mkdir /app && \
-    chown 10001:10001 /app && \
-    groupadd --gid 10001 app && \
-    useradd --no-create-home --uid 10001 --gid 10001 --home-dir /app app
+# Get rustup https://rustup.rs/ for canonicaljson-rs
+# minimal profile https://rust-lang.github.io/rustup/concepts/profiles.html
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --profile minimal -y
+# Add cargo to PATH
+RUN . ~/.cargo/env
+
+RUN python -m venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
 
 COPY requirements.txt .
-COPY bin/docker-install.sh .
-RUN ./docker-install.sh
 
-# TODO: don't copy any /tests folders for prod image (maybe handle this in .dockerignore)
-COPY . /app
+# Python packages
+RUN pip install --upgrade pip
+RUN apt-get install -y --no-install-recommends libpq-dev
+RUN pip install --no-cache-dir -r requirements.txt
+
+RUN uwsgi --build-plugin https://github.com/Datadog/uwsgi-dogstatsd
+
+
+FROM python:3.9-slim-bullseye AS build
```

Comment:
Same comment from above, but just wanted to call out the additional location for a hash. Maybe have them share the hash in some fashion? A Docker `ARG` that could be passed, maybe?

Reply:
> Maybe have them share the hash in some fashion? A Docker ARG that could be passed, maybe?

Normally I'd say this is a good call, but one advantage to not doing this is that Dependabot appears to scan this Dockerfile and [open PRs to upgrade the SHA](https://github.com/mozilla-services/kinto-dist/pull/1912). I'm guessing that we'd lose this if we specified the SHA as a build arg. Since we have Dependabot to open these PRs for us though, that should keep things in sync.